### ISSUE TITLE
Port FxCop rule CA1305: SpecifyIFormatProvider

### ISSUE THREAD

**Issue by lgolding** [dotnet/roslyn-analyzers#413]
**Title:** Specify IFormatProvider

**Description:**

A method or constructor calls one or more members that have overloads that accept a System.IFormatProvider parameter, and the method or constructor does not call the overload that takes the IFormatProvider parameter. When a System.Globalization.CultureInfo or IFormatProvider object is not supplied, the default value that is supplied by the overloaded member might not have the effect that you want in all locales.

**Dependency:** None

**Notes:**

**Comment by srivatsn** [comment]
@qinxgit plans to implement this.

**Comment by srivatsn** [comment]
Removing community tag since @qinxgit is not going to be able to get to this.

**Comment by genlu** [comment]
I have an prototype for this rule here:
https://github.com/genlu/roslyn-analyzers/commit/6382e0cdb2c9dad086eddb596037fceac6e5be0c

Anyone who picks up this work, please feel free to use that code :)

**Comment by basoundr** [comment]
#912 implements this rule

**Comment by basoundr** [comment]
Closed it by mistake. Reopening

**Comment by scott-lin** [comment]
Is this done? I see #912 implements this rule and it is merged into master.

**Comment by mavasani** [comment]
@scott-lin This is due to a bug in the underlying Microsoft.CodeAnalysis package in 2.3 bits. Can you try the latest 2.6.0-beta1 bits on VS 2017.5: https://github.com/dotnet/roslyn-analyzers#recommended-version-of-analyzer-packages

**Comment by mavasani** [comment]
The analyzer for this is done - fixer is still needed.

**Comment by qinxgit** [comment]
Just curious, now the Desktop Analyzers are renamed to “FxCop Analyzers”?

**Comment by mavasani** [comment]
@qinxgit No, we have NetFramework and NetCore analyzers. See https://github.com/dotnet/roslyn-analyzers/blob/master/nuget/Microsoft.CodeAnalysis.FxCopAnalyzers/Microsoft.CodeAnalysis.FxCopAnalyzers.Package.csproj#L15

**Comment by Andrew-Webb** [comment]
Bump.  This needs to happen.  I know too many developers who are completely unaware that, for example, the following can give different results depending on which machine (which culture) it's run on:-

    var val = 3.1415926;
    return val.ToString();

**Comment by mavasani** [comment]
@Andrew-Webb the rule is already implemented, only the code fixer is remaining.

**Comment by Andrew-Webb** [comment]
> only the code fixer is remaining

What does that mean?

**Comment by mavasani** [comment]
> What does that mean?

The diagnostic will fire correctly when you install the package. However, there is no quick action/code fix/light bulb item that can automatically fix the issue. We mark such issues with "Needs-Fixer" tag to track implemented analyzers that need corresponding code fixers.

**Comment by Andrew-Webb** [comment]
Aha!  Another light bulb moment.  This explains why some (other) issues I've seen can't be fixed / suppressed / whatever.  No light bulb!  I thought it was VS flakiness.

(Although I did have a situation where there was no light bulb for an issue for ages, but one eventually appeared.)
